src/agent/cobertura/src/lib.rs

```diff
@@ -17,6 +19,16 @@ impl CoberturaCoverage {
         let text = String::from_utf8(data)?;
         Ok(text)
     }
+
+    pub async fn to_string_async(&self) -> anyhow::Result<String> {
+        let mut buffer = Vec::new();
+        let mut bufwriter = tokio::io::BufWriter::new(&mut buffer);
+
+        self.write_xml_async(&mut bufwriter).await?;
+        bufwriter.flush().await?;
+        let text = String::from_utf8(buffer)?;
+        Ok(text)
+    }
```

Comment:
I wouldn't add this as it's a bit misleading, all the operations are on an in-memory so it doesn't need to be async.